gnome-extensions/test-settings.sh

```
@@ -0,0 +1,248 @@
+    
+    # Remove quotes if present
+    val1=$(echo "$val1" | tr -d '"')
+    val2=$(echo "$val2" | tr -d '"')
+    
+    # Check if values are empty or non-numeric
+    if [ -z "$val1" ] || [ -z "$val2" ]; then
+        return 1
+    fi
+    
+    # Use bc for floating point comparison
+    local result=$(echo "scale=10; (($val1 - $val2) < 0) * (($val2 - $val1) < $epsilon) + (($val1 - $val2) >= 0) * (($val1 - $val2) < $epsilon)" | bc -l 2>/dev/null)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)